dace/transformation/interstate/sdfg_nesting.py

```diff
@@ -455,6 +460,370 @@ def traverse(mtree_node):
+                    out_candidates[e.data.data] = (e.data, nstate,
+                                                   set(range(len(
+                                                       e.data.subset))))
+                for e in nstate.out_edges(dnode):
+                    # If more than one unique element detected, remove from
+                    # candidates
+                    if e.data.data in in_candidates:
+                        memlet, ns, indices = in_candidates[e.data.data]
+                        # Try to find dimensions in which there is a mismatch
+                        # and remove them from list
+                        for i, (s1, s2) in enumerate(
+                                zip(e.data.subset, memlet.subset)):
+                            if s1 != s2 and i in indices:
+                                indices.remove(i)
+                        if len(indices) == 0:
+                            ignore.add(e.data.data)
+                        in_candidates[e.data.data] = (memlet, ns, indices)
+                        continue
+                    in_candidates[e.data.data] = (e.data, nstate,
+                                                  set(range(len(
+                                                      e.data.subset))))
+
+        # TODO: Check in_candidates in interstate edges as well
+
+        # Check in/out candidates
+        for cand in in_candidates.keys() & out_candidates.keys():
+            s1, nstate1, ind1 = in_candidates[cand]
+            s2, nstate2, ind2 = out_candidates[cand]
+            indices = ind1 & ind2
+            if any(s1.subset[ind] != s2.subset[ind] for ind in indices):
+                ignore.add(cand)
+            in_candidates[cand] = (s1, nstate1, indices)
+            out_candidates[cand] = (s2, nstate2, indices)
+
+        # Ensure minimum elements of candidates do not begin with zero
+        def _check_cand(candidates, outer_edges):
+            for cname, (cand, nstate, indices) in candidates.items():
+                if all(me == 0 for i, me in enumerate(cand.subset.min_element())
+                       if i in indices):
+                    ignore.add(cname)
+                    continue
+
+                # Ensure outer memlets begin with 0
+                outer_edge = next(iter(outer_edges(nsdfg, cname)))
+                if any(me != 0 for i, me in enumerate(
+                        outer_edge.data.subset.min_element()) if i in indices):
+                    ignore.add(cname)
+                    continue
+
+                # Check w.r.t. loops
+                if len(nstate.ranges) > 0:
+                    # Re-annotate loop ranges, in case someone changed them
+                    # TODO: Move out of here!
+                    nstate.ranges = {}
+                    from dace.sdfg.propagation import _annotate_loop_ranges
+                    _annotate_loop_ranges(nsdfg.sdfg, [])
+                    #propagation.propagate_states(nsdfg.sdfg)
```

Comment:
```suggestion
                    # Re-annotate loop ranges, in case someone changed them
                    # TODO: Move out of here!
                    nstate.ranges = {}
                    from dace.sdfg.propagation import _annotate_loop_ranges
                    _annotate_loop_ranges(nsdfg.sdfg, [])
```

Is this a TODO(later)?

Reply:
Yes